packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -111,10 +127,25 @@ export async function prefetchLatestSnapshot(
 				putInCache,
 				removeEntries,
 				enableRedeemFallback,
-			);
+			)
+				.then((value) => {
+					assert(!!snapshotEpoch, "prefetched snapshot should have a valid epoch");
+					snapshotContentsWithEpochP.resolve({ ...value, fluidEpoch: snapshotEpoch });
+				})
+				.catch((err) => {
+					snapshotContentsWithEpochP.reject(err);
+					// Remove from cache if prefetch was not successful.
+					snapshotNonPersistentCache?.remove(nonPersistentCacheKey);
+					throw err;
+				});
 			assert(cacheP !== undefined, 0x1e7 /* "caching was not performed!" */);
-			await cacheP;
+			// Remove it from the non persistent cache once it is cached in the persistent cache or an error
+			// occured while caching.
+			await cacheP.finally(() => snapshotNonPersistentCache?.remove(nonPersistentCacheKey));
```

Comment:
It's a bit hard to follow code.
I wonder if you could move "await cacheP" after line 133 (to successful path of prefetching), and thus do snapshotNonPersistentCache?.remove() in .finally block to fetchSnapshotWithRedeem().

I.e.

```ts
fetchSnapshotWithRedeem()
    .then(() => {
        ...
        await cacheP;
    }.catch(() => {
        ...
    }.finally (() => {
        snapshotNonPersistentCache?.remove
    }
}

Reply:
Done